external-crates/move/crates/move-coverage/src/lcov.rs

```diff
@@ -0,0 +1,368 @@
+                    record
+                        .functions_hit
+                        .entry(name.clone())
+                        .and_modify(|e| *e += 1)
+                        .or_insert(1);
+                    let Ok(smap) =
+                        record
+                            .unit
+                            .source_map
+                            .get_function_source_map(FunctionDefinitionIndex(
+                                frame.binary_member_index,
+                            ))
+                    else {
+                        continue;
+                    };
+                    let line = self
+                        .file_mapping
+                        .start_position(&smap.definition_location)
+                        .line_offset()
+                        + 1;
+                    record
+                        .line_entries
+                        .entry(line)
+                        .and_modify(|e| *e += 1)
+                        .or_insert(1);
+                    current_fn_index.push(frame.binary_member_index);
+                    current_record_id.push(module_id);
+                    coming_from = None;
+                }
+                TraceEvent::Instruction { pc, .. } => {
+                    let module_id = current_record_id.last().unwrap();
+                    let current_fn_index = current_fn_index.last().unwrap();
+                    let record = self.file_record_keepers.get_mut(module_id).unwrap();
+                    let Ok(loc) = record
+                        .unit
+                        .source_map
+                        .get_code_location(FunctionDefinitionIndex(*current_fn_index), pc)
+                    else {
+                        continue;
+                    };
+                    let line = self.file_mapping.start_position(&loc).line_offset() + 1;
+                    debug_assert!(record.instrumented_lines.contains(&line));
+
+                    record
+                        .line_entries
+                        .entry(line)
+                        .and_modify(|e| *e += 1)
+                        .or_insert(1);
+
+                    if let Some(from) = coming_from {
+                        if let Some(info) = record.branches.get_mut(&(*current_fn_index, from)) {
+                            info.hit_branch(pc)
+                        }
+                    }
+
+                    coming_from = None;
+
+                    let branch_info_opt = record.branches.get_mut(&(*current_fn_index, pc));
+
+                    if branch_info_opt.is_some() {
+                        coming_from = Some(pc);
+                    }
+                }
+                TraceEvent::CloseFrame { .. } => {
+                    current_fn_index.pop();
+                    current_record_id.pop();
+                    coming_from = None;
+                }
+                TraceEvent::Effect(_) | TraceEvent::External(_) => (),
```

Comment:
Do you need to set `coming_from = None` here?

Reply:
(I don't really know or understand what these trace event cases do)

Reply:
Good question -- we shouldn't set it here as `Effect` events simply record effects of an `Instruction` event (so have no impact on the CFG), and likewise `External` events are completely opaque and similarly have no impact on the CFG.